### DISCUSSION TITLE
Is instance the same as gdsfactory.ComponentReference

### DISCUSSION THREAD

**Discussion by DanBThom** [gdsfactory/gdsfactory#3436] [Q&A] [answered]
loading a gds file and attempting to print an instance name, as well as the parent component gives an error:
AttributeError: 'Instance' object has no attribute 'parent'
the gdsfactory.ComponentReference attributes specify a parent attribute hence my confusion

import gdsfactory as gf
gdspath = 'myfile.gds'
c_gf = gf.import_gds(gdspath)

for instance in c_gf.insts:  #an instance is a Component_reference
    print(instance.name)
    print(instance.parent)    
c_gf.write_gds("mylayout1.gds")
c_gf.plot()

**Comment by joamatab** [comment, marked as answer]
Here’s how you can correctly access the information:

```python
import gdsfactory as gf

gdspath = 'myfile.gds'
c_gf = gf.import_gds(gdspath)

for instance in c_gf.insts:  # an instance is a ComponentReference
    print(instance.cell.name)
```

**Comment by DanBThom** [comment]
Thanks Joamatab, the instance.name and instance.cell.name both have no errors, but the instance.parent does.  Why is that?

**Reply by joamatab** [reply]
instance.parent is a property managed internally by KLayout. Modifying or relying on it directly can lead to unintended behavior, so it is recommended to avoid using it.
